sdk/core/core-util/src/aborterUtils.ts

```diff
@@ -61,3 +71,23 @@ export function createAbortablePromise<T>(
     abortSignal?.addEventListener("abort", onAbort);
   });
 }
+
+/**
+ * promise.race() implementation that aborts losers as soon as the first promise fulfills.
+ */
+export async function racePromisesAndAbortLosers<T>(
+  promises: ((abortOptions: AbortOptions) => Promise<T>)[],
+  abortSignal?: AbortSignalLike
```

Comment:
Can we put this into an options bag to future-proof adding more options in the future?